diff --git a/services/java/com/android/server/wm/StackTapPointerEventListener.java b/services/java/com/android/server/wm/StackTapPointerEventListener.java
index 19d8ab3..4831c67 100644
--- a/services/java/com/android/server/wm/StackTapPointerEventListener.java
+++ b/services/java/com/android/server/wm/StackTapPointerEventListener.java
@@ -16,6 +16,7 @@
 
 package com.android.server.wm;
 
+import android.util.Slog;
 import android.graphics.Region;
 import android.view.DisplayInfo;
 import android.view.MotionEvent;
@@ -26,6 +27,8 @@ import com.android.server.wm.WindowManagerService.H;
 public class StackTapPointerEventListener implements PointerEventListener {
     private static final int TAP_TIMEOUT_MSEC = 300;
     private static final float TAP_MOTION_SLOP_INCHES = 0.125f;
+    private static final String TAG = "StackTapPointerEventListener";
+    private static final boolean DEBUG = false;
 
     private final int mMotionSlop;
     private float mDownX;
@@ -74,7 +77,8 @@ public class StackTapPointerEventListener implements PointerEventListener {
                     if ((motionEvent.getEventTime() - motionEvent.getDownTime())
                             < TAP_TIMEOUT_MSEC
                             && (x - mDownX) < mMotionSlop && (y - mDownY) < mMotionSlop
-                            && !mTouchExcludeRegion.contains(x, y)) {
+                            && !mTouchExcludeRegion.contains(x, y)
+                            && !inInputMethodUi(x, y)) {
                         mService.mH.obtainMessage(H.TAP_OUTSIDE_STACK, x, y,
                                 mDisplayContent).sendToTarget();
                     }
@@ -84,4 +88,15 @@ public class StackTapPointerEventListener implements PointerEventListener {
             }
         }
     }
+
+    private boolean inInputMethodUi(int x, int y) {
+        WindowState state = this.mService.mInputMethodWindow;
+        if (state == null) {
+            return false;
+        }
+        Region region = new Region();
+        state.getTouchableRegion(region);
+        return region.contains(x, y);
+    }
+
 }

